packages/foam-core/src/note-graph.ts

```diff
@@ -91,9 +94,7 @@ export class NoteGraph implements NoteGraphAPI {
     const note = this.getNote(noteId);
     if (isSome(note)) {
       this.graph.removeNode(noteId);
```

Comment:
this would cause the inbound/outbound links to be removed, no? 
I reckon we would need to set it to `null` to keep the existing inbound connections

Reply:
It looks like `doDelete` is now only being called when a note is deleted which I think means this shouldn't be a problem anymore.
![image](https://user-images.githubusercontent.com/4162073/100903943-781e7c00-3494-11eb-8f4f-ae4f912ce677.png)

If there's a concern about a deletion rippling to connections between neighboring files, those seem fine.

<img width="244" alt="Screen Shot 2020-12-02 at 11 51 57 AM" src="https://user-images.githubusercontent.com/4162073/100904422-02ff7680-3495-11eb-8545-0fb1e93debf4.png">

Here if I delete "types of food" the original connections in "soup" remain
<img width="198" alt="Screen Shot 2020-12-02 at 11 52 26 AM" src="https://user-images.githubusercontent.com/4162073/100904509-16aadd00-3495-11eb-9287-bfa62ca520de.png">

Reply:
if `pasta` has a connection to `types-of-food`, I would expect to see that connection even afterwards, but instead of being associated to a real note, it would be associated to a placeholder note

Reply:
Ah I see. 

So... On delete, if a note has inbound links, do not entirely remove the node just set it back to a placeholder. 

This seems to be an existing problem with the current release. 

Mind if I add that to the issue tracker and try to address that separately? It doesn't look like it'll overlap with this change to update events.

I'm also having some recurring issues running the extension in dev mode that are making it a little tricky, I'll ask about those in Discord later and see if it's anything I can update the docs for as well.

Reply:
I understand that your specific issue was with the removal of the inbound links, which you have now solved.
I see this issue as fixing a problem with the delete function, which has not been solved yet. 
Is there a reason why you'd rather not make the change here?

Also, it's important to add the test cases for:
- [ ] the bug that was fixed
- [ ] the current incorrect behavior (the fact that tests are not failing now means we are not capturing this case)

I am also happy to contribute to the branch, for me what matters is merging a complete fix to the deletion bug.